nmrcraft/data/dataset.py

```diff
@@ -278,6 +278,35 @@ def more_than_one_target(self):
         """Function returns true if more than one target is specified"""
         return len(self.target_columns) > 1
 
+    def categorical_target_decoder(self, y):
+        """
+        function takes in the  target (y) array and transforms it back to decoded form.
+        For this function to be run the split_and_preprocess_categorical already has to have been run beforehand.
+        """
+        ys = y[:]  # copy y so it's not modified
+        target_encoders = self.target_label_encoders
+        ys_decoded = []
+        ys = [  # rotate the y array to each column be a target
+            list(x) if i == 0 else x
+            for i, x in enumerate(
+                map(list, zip(*ys))
+            )  # y[:] should leave original y in peace?
+        ]
+
+        # Decode columnwise
+        for i, target_column in enumerate(ys):
+            ys_decoded.append(
+                target_encoders[i].inverse_transform(target_column)
+            )
+
+        # Rotate back so each row corresponds to a complex and not the target like metal or X4
+        ys_decoded_properly_rotated = [
+            list(x) if i == 0 else x
+            for i, x in enumerate(map(list, zip(*ys_decoded)))
+        ]
+
+        return ys_decoded_properly_rotated
```

Comment:
Nice, thank you for the addition and testing it with the scikit learn confusion matrix, that's super valuable! I think the code needs more commenting to make 100% clear what each function/part does.

Reply:
e.g., what does decode, rotate, binarize, ... mean. We can add this as docstrings with examples to the function definition

Reply:
I will take care of the whole rotatation stuff when refactoring. I already have a decent idea of how to take the code apart to make it clearer. But I definetly agree that "rotate" makes it hard to understand 😆